src/Containers.jsx

```diff
@@ -198,6 +230,42 @@ class Containers extends React.Component {
                 });
     }
 
+    handleCheckpointContainer(args) {
+        const container = this.state.containerWillCheckpoint;
+        this.setState((prevState) => ({
+            selectContainerCheckpointModal: false
+        }));
+        client.postContainer(container.isSystem, "checkpoint", container.Id, args)
```

Comment:
In trivial cases both checkpoint and restore operations are very fast. But if the container in interest is running some memory heavy operation when we want to checkpoint that it might take some time to store all these into files right?

So I think we should keep the modal open and only close that when the operation is really finished.
We should show a spinner, disable the Checkpoint button etc.
Something you can achieve with the diff below:
``` diff
diff --git src/ContainerCheckpointModal.jsx src/ContainerCheckpointModal.jsx
index d44bdee8..e5bf3d83 100644
--- src/ContainerCheckpointModal.jsx
+++ src/ContainerCheckpointModal.jsx
@@ -1,6 +1,6 @@
 import React from 'react';
 import { Modal } from 'patternfly-react';
-import { Button, Checkbox } from '@patternfly/react-core';
+import { Button, Checkbox, Spinner } from '@patternfly/react-core';
 import cockpit from 'cockpit';
 import * as utils from './util.js';
 
@@ -48,8 +48,8 @@ class ContainerCheckpointModal extends React.Component {
                     </div>
                 </Modal.Body>
                 <Modal.Footer>
-                    <Button variant="primary" onClick={() => this.props.handleCheckpointContainer(this.state)}>
-                        {_("Checkpoint")}
+                    <Button variant="primary" isDisabled={this.props.checkpointInProgress} onClick={() => this.props.handleCheckpointContainer(this.state)}>
+                        {!this.props.checkpointInProgress ? _("Checkpoint") : <Spinner size="sm" className="pull-right" />}
                     </Button>
                     <Button variant="link" onClick={this.props.handleCheckpointContainerDeleteModal}>
                         {_("Cancel")}
diff --git src/Containers.jsx src/Containers.jsx
index dbdcded0..7a70c564 100644
--- src/Containers.jsx
+++ src/Containers.jsx
@@ -232,13 +232,14 @@ class Containers extends React.Component {
 
     handleCheckpointContainer(args) {
         const container = this.state.containerWillCheckpoint;
-        this.setState((prevState) => ({
-            selectContainerCheckpointModal: false
-        }));
+        this.setState({ checkPointInProgress: true });
         client.postContainer(container.isSystem, "checkpoint", container.Id, args)
                 .catch(ex => {
                     const error = cockpit.format(_("Failed to checkpoint container $0"), container.Names);
                     this.props.onAddNotification({ type: 'danger', error, errorDetail: ex.message });
+                })
+                .finally(() => {
+                    this.setState({ selectContainerCheckpointModal: false, checkPointInProgress: false });
                 });
     }
 
@@ -333,6 +334,7 @@ class Containers extends React.Component {
                 selectContainerCheckpointModal={this.state.selectContainerCheckpointModal}
                 handleCheckpointContainer={this.handleCheckpointContainer}
                 handleCheckpointContainerDeleteModal={this.handleCheckpointContainerDeleteModal}
+                checkPointInProgress={this.state.checkPointInProgress}
                 containerWillCheckpoint={this.state.containerWillCheckpoint}
             />;
         const containerRestoreModal =

```

Let me know if you have some objections here or you prefer that I implement it as a followup.

Reply:
That's a good point, checkpoint/restore can take longer on heavier containers (I only tested it on a simple Node.js counter) - I'll implement the spinner.